### DISCUSSION TITLE
Get body overlapping pairs

### DISCUSSION THREAD

**Discussion by vsirvent** [DanielChappuis/reactphysics3d#319] [Q&A]
Hi,

First of all. Thank you very much for the engine. I like the source code and how easy is to use it.

In order to modify the path of my creatures when they collide during movement, I want to get current body contacts with other bodies of the world when the creature is moving and the only way I can find is iterating all the overlapping contact pairs of the world, which is very heavy as every body is in contact with the floor and I need to check it very often. Is there any way to get the current contacts of a specific body?

**Comment by DanielChappuis** [comment]
Hello,

Thanks a lot for your feedback.

You should take a look at the [Receiving feedback](https://reactphysics3d.com/usermanual.html#x1-7600015) section in the documentation. It explains how to inherit from the EventListener class of the library to receive callbacks when contacts occur in your simulation.

I hope this helps.
